### ISSUE TITLE
Windows build fails with special characters in path

### ISSUE THREAD

**Issue by m-allanson** [gatsbyjs/gatsby#4565]
## Description

Builds will fail on Windows for paths that contain non-`[a-zA-Z0-9]` characters such as `ä`, `ö`, `ü`.

Refs #3183 @anttti @LeKoArts 

### Steps to reproduce

- On a Windows PC
- Create a Gatsby project in a folder who's path contains characters like `ä`, `ö`, `ü`
- Run `gatsby build`

### Expected result

The site should build

### Actual result

The site errors with:

```
error UNHANDLED REJECTION

  TypeError: Cannot read property 'jsonName' of undefined
```

### Environment

* Operating System: Windows

**Comment by KyleAMathews** [comment]
Due to the high volume of issues, we're closing out older ones without recent activity. Please open a new issue if you need help!

**Comment by LeKoArts** [comment]
@KyleAMathews Don’t understand why this one is closed... It’s still a bug at least 6 people had and it just wasn’t solved by anyone until now

**Comment by KyleAMathews** [comment]
👍 cool — just closed 400 so might have closed a few that shouldn't have.

**Comment by elitan** [comment]
How do I do links with special chars? Ex links included: `å`, `ä` and `ö` (Swedish special chars)?

**Comment by vladar** [comment]
I am seeing different issue related to GraphQL:

```
The GraphQL query from C:/dev/sandbox/path-with-ä-ру/src/components/image.js failed.

Errors:
  Syntax Error: Cannot parse the unexpected character "\u0420".

  GraphQL request:1:27
  1 | query cDevSandboxPathWithAРуSrcComponentsImageJs2969191536 {
    |                           ^
  2 |   placeholderImage: file(relativePath: {eq: "gatsby-astronaut.png"}) {
Plugin:
  none
Query:
  query cDevSandboxPathWithAРуSrcComponentsImageJs2969191536 {
    placeholderImage: file(relativePath: {eq: "gatsby-astronaut.png"}) {
      childImageSharp {
        fluid(maxWidth: 300) {
          ...GatsbyImageSharpFluid
        }
      }
    }
  }

  fragment GatsbyImageSharpFluid on ImageSharpFluid {
    base64
    aspectRatio
    src
    srcSet
    sizes
  }
```

**Comment by wardpeet** [comment]
Feels like we need to slugify the query name. Does that sound right @vladar ?

**Comment by vladar** [comment]
Yep, do we use any library for this already? Or maybe you have one in mind?

**Comment by vladar** [comment]
Should be fixed in gatsby `2.17.17`
